### PR TITLE
activate scalene and bump to latest version

### PR DESCRIPTION
**Pull request by tanaysd**
@ianspektor @javiber ptal, all bazel tests passed

### CODE REVIEW

**Comment by javiber** [comment]
Hi @tanaysd thanks for open this PR so soon!

The tests are complaining that `poetry.lock` is out of sync with the `pyproject.toml` can you include the lock file?

**Comment by ianspektor** [comment]
Thanks for taking on this @tanaysd! Note also that the benchmarks aren't ran as part of the test suite but rather need to be ran manually locally to ensure they work, see [this comment] in the original issue.

**Comment by tanaysd** [comment]
@ianspektor are you referring to [this comment](https://github.com/google/temporian/issues/354#issuecomment-1899159416)?

@javiber will update `.lock` file as well

**Comment by tanaysd** [comment]
@javiber PTAL, update `lock` file, added python restrictions for Scalene as indicated by Scalene's author

@ianspektor executred `bazel` tests for three scripts in the script folder, what should I be looking for?

**Comment by ianspektor** [comment]
> @ianspektor are you referring to https://github.com/google/temporian/issues/354#issuecomment-1899159416?

yep, sorry :)

> @ianspektor executed bazel tests for three scripts in the script folder, what should I be looking for?

if `bazel run -c opt --config=linux //benchmark:profile_time -- basic` (or `-- sma` or `-- add_index`) runs and shows scalene's output you're good to go! let us know so we can test it locally before merging. thanks!

**Comment by tanaysd** [comment]
> > @ianspektor are you referring to [#354 (comment)](https://github.com/google/temporian/issues/354#issuecomment-1899159416)?
> 
> yep, sorry :)
> 
> > @ianspektor executed bazel tests for three scripts in the script folder, what should I be looking for?
> 
> if `bazel run -c opt --config=linux //benchmark:profile_time -- basic` (or `-- sma` or `-- add_index`) runs and shows scalene's output you're good to go! let us know so we can test it locally before merging. thanks!

Yes, for all three tests, it opened a local webpage displaying `scalene` metrics -- you are good to test locally.

**Comment by ianspektor** [comment]
Confirm that scalene installs and runs 👍🏼 Thanks!

Some stuff seems to need to be fixed/updated for Scalene's new version (maybe @emeryberger can further comment on these):

- Running `bazel run -c opt --config=linux benchmark:profile_time -- sma` now shows no line output, only the total result for the `main()` function in `benchmark/scripts/sma.py`. I *think* it showed line-by-line results for the function annotated with @profile before. I'd remove the `@profile` annotation now, which shows the full output (we were using it before because some imports were heavy enough to contaminate the output, which is still the case, but that's better than not seeing the line-by-line).
image

- I'd remove the @profile annotation from the other scripts in `benchmark/scripts` too.
- The `-p` option doesn't seem to exist anymore, it needs to be removed from the `### Benchmarking and profiling` section in `CONTRIBUTING.md`
- The `benchmark/scripts/basic.py` script fails when ran, it's using `EventSet.first_index_data()` that was deprecated. Printing the whole EventSet as a fix is fine.

**Comment by tanaysd** [comment]
> Confirm that scalene installs and runs 👍🏼 Thanks!
> 
> Some stuff seems to need to be fixed/updated for Scalene's new version (maybe @emeryberger can further comment on these):
> 
> * Running `bazel run -c opt --config=linux benchmark:profile_time -- sma` now shows no line output, only the total result for the `main()` function in `benchmark/scripts/sma.py`. I _think_ it showed line-by-line results for the function annotated with @Profile before. I'd remove the `@profile` annotation now, which shows the full output (we were using it before because some imports were heavy enough to contaminate the output, which is still the case, but that's better than not seeing the line-by-line).
> 
> image
> * I'd remove the @Profile annotation from the other scripts in `benchmark/scripts` too. -- `DONE`
> * The `-p` option doesn't seem to exist anymore, it needs to be removed from the `### Benchmarking and profiling` section in `CONTRIBUTING.md` -- `DONE`
> * The `benchmark/scripts/basic.py` script fails when ran, it's using `EventSet.first_index_data()` that was deprecated. Printing the whole EventSet as a fix is fine. -- `DONE`

@ianspektor @javiber PTAL

File: CONTRIBUTING.md
Comment (ianspektor) [inline CONTRIBUTING.md:98]: @tanaysd - big mistake on my side, `profile_memory` uses `mprof`, not scalene (see `benchmark/profile_memory.sh`) 😬 please revert this change. LGTM from my side after that! 👍🏼

Comment (tanaysd) [reply CONTRIBUTING.md]: done.

**Review by ianspektor — APPROVED** [review]

**Comment by ianspektor** [comment]
LGTM - thanks @tanaysd!

**Comment by emeryberger** [comment]
The only reason I can think that there wouldn't be line level information is that no individual line accounted for more than 1% of execution time (or memory, if you are profiling that). That logic is not affected by the presence or absence of the `@profile` decorator. It would make sense to look at the `profile.json` file to verify that individual lines are being tracked.

**Pull request merged**
